posts/2022-03-01-user-feature-install.adoc

```diff
@@ -0,0 +1,378 @@
+
+A Liberty feature requires the following headers:
+
+- *Subsystem-SymbolicName* : Specifies the identity and visibility of the feature; 
+- *Subsystem-Content* : Comma-separated list of bundles and subsystems that are required to run this feature;
+- *IBM-Feature-Version* : Identifies which version of feature support is required by the runtime environment. Must be set to 2;
+- *Subsystem-Type* : All features are currently of the same subsystem type `osgi.subsystem.feature`;
+
+For details about the format of a feature manifest file, see link:https://www.ibm.com/docs/en/was-liberty/base?topic=manually-liberty-feature-manifest-files[Liberty feature manifest files].
+
+We will use the link:https://aries.apache.org/documentation/modules/esamavenpluginproject.html[esa-maven-plugin] to package our bundle and to generate SUBSYSTEM.MF file. Set the `packaging` type to *esa* and add the OSGi bundle dependency and appropriate headers. 
+
+.pom.xml
+[source, xml]
+----
+<project>
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>test.user.test.osgi</groupId>
+  <artifactId>SimpleActivatorESA-</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+
+  <packaging>esa</packaging> <!-- set packaging type to esa -->
+
+  <dependencies>
+    <!-- Add OSGi bundle -->
+    <dependency>
+		<groupId>test.user.test.osgi</groupId>
+    	<artifactId>SimpleActivator</artifactId>
+    	<version>0.0.1-SNAPSHOT</version>
+	</dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.aries</groupId>
+        <artifactId>esa-maven-plugin</artifactId>
+        <version>1.0.0</version>
+        <extensions>true</extensions>
+        <configuration>
+          <generateManifest>true</generateManifest>
+          <archiveContent>all</archiveContent>
+          <instructions>
+            <Subsystem-Vendor>IBM</Subsystem-Vendor>
+            <IBM-Feature-Version>2</IBM-Feature-Version>
+            <IBM-ShortName>SimpleActivator</IBM-ShortName>
+            <Subsystem-Type>osgi.subsystem.feature</Subsystem-Type>
+            <Subsystem-SymbolicName>
+                com.example.bundle.Activator;visibility:=public
+            </Subsystem-SymbolicName>
+            <Subsystem-Version>1.0.0</Subsystem-Version>
+          </instructions>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+</project>
+----
+By default it will not generate a manifest, so we have to set the `generateManifest` header to *true*. Note that in order to install the feature, we need to set the *visibility* directive to "public". We can do so by setting the `Subsystem-SymbolicName` header to "_Bundle-SymbolicName;visibility:=public_". If the *visibility* is set to `protected|private`, Liberty Maven Plugin will not be able to resolve the feature. Also, the mandatory header `Subsystem-Content` will be created automatically by the plugin. 
```

Comment:
it might be helpful to call out `<IBM-ShortName>SimpleActivator</IBM-ShortName>` in the explanation since that value is what goes into the server.xml later on.